src/typeUtilities.ts

```diff
@@ -55,8 +55,11 @@ export type SafeOmit<T, Key extends keyof T> = IsEmpty<
 /**
  * @internal
  */
-export type DistributiveOmit<T, K extends keyof T> = T extends any
-  ? SafeOmit<T, K>
+export type DistributiveOmit<T, K extends keyof T> = T extends T
+  ? // Typescript actually is fine with Omit<T, K>, but this is surprising because
+    // K might include elements that are not in every member of the union T. In other words,
+    // K does not extend keyof T.
+    Omit<T, K & keyof T>
```

Comment:
SafeOmit results in never when it really should be empty here. It wasn't used so i changed it to work.

Reply:
We could do never if you preferred it but it means people can't pass an empty props object which we were doing in one test